src/DssProxyActionsCropper.sol

```diff
@@ -773,15 +772,15 @@ contract DssProxyActionsEndCharter is Common {
         bytes32 ilk = GemJoinLike(gemJoin).ilk();
         EndLike(end).cash(ilk, wad);
         uint256 wadC = _mul(wad, EndLike(end).fix(ilk)) / RAY;
-        // Flux to the proxy's UrnProxy in charter, so it can be pulled out with the managed gem join
+        // Flux to the proxy's UrnProxy in cropper, so it can be pulled out with the managed gem join
         VatLike(vat).flux(
             ilk,
             address(this),
-            CharterLike(charter).getOrCreateProxy(address(this)),
+            CropperLike(cropper).getOrCreateProxy(address(this)),
             wadC
         );
         // Exits token amount to the user's wallet as a token
         uint256 amt = wadC / 10 ** (18 - GemJoinLike(gemJoin).dec());
-        CharterLike(charter).exit(gemJoin, msg.sender, amt);
+        CropperLike(cropper).exit(gemJoin, msg.sender, amt); // TODO: should this change to 2 stages as well?
```

Comment:
As I said above, I'm not sure the reasoning to not have all the bonus tokens go out to the EOA instead of the proxy.

Reply:
I think ideally we want to have the rewards either sent to the owner (EOA) on every action that crops, or kept in the ds-proxy after each such action.
The point is I don't think we should mix between the two.

Note that every time you join gems CropJoin sends out the rewards to the ds-proxy (aka `usr`).
So if we go with the paradigm of always retreiving the rewards to the owner, we have to add special logic there (on several joining actions) to check the bonus balance difference and send it out.
Also on exiting eth we will have to do such balance diff check and send the bonus our explicitly (since we do it in 2 stages for unwrapping weth).
So I think the current implementation is just simpler.
Note that we actually asked the UI teams and they preffered it this way. 

The only potential problem I see with this mode of work is that if we support allowance in proxy-actions level, an allowed actor will be able to move rewards to the ds-proxy, but not pull it out.
For that we might want to add the allow condition also for an action that pulls out rewards in the future, although if not done atomically then another allowed actor (or the owner) can get the rewards in between.
This is something to think about for all reviewers of this PR.

Reply:
> Note that we actually asked the UI teams and they preffered it this way.

Ok this is the most important piece. If this is what they want then sure.

Reply:
>The only potential problem I see with this mode of work is that if we support allowance in proxy-actions level, an allowed actor will be able to move rewards to the ds-proxy, but not pull it out.
For that we might want to add the allow condition also for an action that pulls out rewards in the future, although if not done atomically then another allowed actor (or the owner) can get the rewards in between.
This is something to think about for all reviewers of this PR.

So the use case for this would be something like a user has granted a Vault manager service an allowance on their CDP, and they want that service to also be able do things with their rewards? E.g. sell rewards to cover debt or reinvest in the Vault in some way, etc. Seem like one option for them in that case would be to just grant an allowance for the `bonus` token from their proxy to the service. That wouldn't require any change to this code. I'd favor keeping things simple here (i.e. leaving as they are currently) unless there's a strong motivation for this additional functionality.

Reply:
Yes, the scenario you described is a good exmple. Note that at this point it's a tehoretic scenario since allowance is not supported in this proxy actions contract (in contrary to the original proxy actions). My goal was to think of consequences for when we do support allowance, as at that point changing the way we handle rewards might be not conveinient for integrations.

So the question is whether we think there would be a scenario where we only want one allowed party to be able to get the rewards and not another allowed party. Or we want all allowed partites but not the original owner.
These are probably not likely cases, but we want to understand if they are deal breakers here.

Reply:
Hmmmm ofc the downside to the approach I suggest is that if the user ever has `bonus` tokens stored in their proxy that didn't come out of crop rewards, the manager might (either intentionally or unintentionally) do something with those tokens against the user's will. So if that's a concern you'd want the more limited model of something built in the proxy actions and/or cropper.

> So the question is whether we think there would be a scenario where we only want one allowed party to be able to get the rewards and not another allowed party. Or we want all allowed partites but not the original owner.
These are probably not likely cases, but we want to understand if they are deal breakers here.

If we start with an authorization model that is more permissive (e.g. all allowed parties have access to rewards), then a more granular one can still be built on-top, but not vice-versa (see: the Vat's permission design).